{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.1",
    "cveMetadata": {
        "cveId": "CVE-2025-8291",
        "assignerOrgId": "28c92f92-d60d-412d-b760-e73465c3df22",
        "state": "PUBLISHED",
        "assignerShortName": "PSF",
        "dateReserved": "2025-07-28T21:05:06.237Z",
        "datePublished": "2025-10-07T18:10:05.908Z",
        "dateUpdated": "2025-10-15T15:55:22.109Z"
    },
    "containers": {
        "cna": {
            "affected": [
                {
                    "defaultStatus": "unaffected",
                    "product": "CPython",
                    "repo": "https://github.com/python/cpython",
                    "vendor": "Python Software Foundation",
                    "versions": [
                        {
                            "version": "0",
                            "lessThan": "3.9.24",
                            "status": "affected",
                            "versionType": "python"
                        },
                        {
                            "version": "3.10.0",
                            "lessThan": "3.10.19",
                            "status": "affected",
                            "versionType": "python"
                        },
                        {
                            "version": "3.11.0",
                            "lessThan": "3.11.14",
                            "status": "affected",
                            "versionType": "python"
                        },
                        {
                            "version": "3.12.0",
                            "lessThan": "3.12.12",
                            "status": "affected",
                            "versionType": "python"
                        }
                    ]
                }
            ],
            "credits": [
                {
                    "lang": "en",
                    "type": "reporter",
                    "value": "Caleb Brown (Google)"
                },
                {
                    "lang": "en",
                    "type": "remediation developer",
                    "value": "Serhiy Storchaka"
                },
                {
                    "lang": "en",
                    "type": "coordinator",
                    "value": "Seth Larson"
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "supportingMedia": [
                        {
                            "base64": false,
                            "type": "text/html",
                            "value": "<p>The 'zipfile' module would not check the validity of the ZIP64 End of\nCentral Directory (EOCD) Locator record offset value would not be used to\nlocate the ZIP64 EOCD record, instead the ZIP64 EOCD record would be\nassumed to be the previous record in the ZIP archive. This could be abused\nto create ZIP archives that are handled differently by the 'zipfile' module\ncompared to other ZIP implementations.</p>\n<p>Remediation maintains this behavior, but checks that the offset specified\nin the ZIP64 EOCD Locator record matches the expected value.</p>"
                        }
                    ],
                    "value": "The 'zipfile' module would not check the validity of the ZIP64 End of\nCentral Directory (EOCD) Locator record offset value would not be used to\nlocate the ZIP64 EOCD record, instead the ZIP64 EOCD record would be\nassumed to be the previous record in the ZIP archive. This could be abused\nto create ZIP archives that are handled differently by the 'zipfile' module\ncompared to other ZIP implementations.\n\n\nRemediation maintains this behavior, but checks that the offset specified\nin the ZIP64 EOCD Locator record matches the expected value."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "NONE",
                        "baseScore": 4.3,
                        "baseSeverity": "MEDIUM",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "LOW",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "REQUIRED",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:L/A:N",
                        "version": "3.1"
                    },
                    "format": "CVSS",
                    "scenarios": [
                        {
                            "lang": "en",
                            "value": "GENERAL"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "orgId": "28c92f92-d60d-412d-b760-e73465c3df22",
                "shortName": "PSF",
                "dateUpdated": "2025-10-14T15:58:46.548Z"
            },
            "references": [
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/pull/139702"
                },
                {
                    "tags": [
                        "vendor-advisory"
                    ],
                    "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/QECOPWMTH4VPPJAXAH2BGTA4XADOP62G/"
                },
                {
                    "tags": [
                        "issue-tracking"
                    ],
                    "url": "https://github.com/python/cpython/issues/139700"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/162997bb70e067668c039700141770687bc8f267"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/333d4a6f4967d3ace91492a39ededbcf3faa76a6"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/1d29afb0d6218aa8fb5e1e4a6133a4778d89bb46"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/76437ac248ad8ca44e9bf697b02b1e2241df2196"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/8392b2f0d35678407d9ce7d95655a5b77de161b4"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/bca11ae7d575d87ed93f5dd6a313be6246e3e388"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/d11e69d6203080e3ec450446bfed0516727b85c3"
                }
            ],
            "source": {
                "discovery": "UNKNOWN"
            },
            "title": "ZIP64 End of Central Directory (EOCD) Locator record offset not checked",
            "x_generator": {
                "engine": "Vulnogram 0.2.0"
            }
        },
        "adp": [
            {
                "problemTypes": [
                    {
                        "descriptions": [
                            {
                                "type": "CWE",
                                "cweId": "CWE-1285",
                                "lang": "en",
                                "description": "CWE-1285 Improper Validation of Specified Index, Position, or Offset in Input"
                            }
                        ]
                    }
                ],
                "metrics": [
                    {
                        "other": {
                            "type": "ssvc",
                            "content": {
                                "timestamp": "2025-10-08T18:41:04.198985Z",
                                "id": "CVE-2025-8291",
                                "options": [
                                    {
                                        "Exploitation": "none"
                                    },
                                    {
                                        "Automatable": "no"
                                    },
                                    {
                                        "Technical Impact": "partial"
                                    }
                                ],
                                "role": "CISA Coordinator",
                                "version": "2.0.3"
                            }
                        }
                    }
                ],
                "title": "CISA ADP Vulnrichment",
                "providerMetadata": {
                    "orgId": "134c704f-9b21-4f2e-91b3-4a467353bcc0",
                    "shortName": "CISA-ADP",
                    "dateUpdated": "2025-10-15T15:55:22.109Z"
                }
            }
        ]
    }
}